TheEthicalHackerCup/Assets/Scripts/GameManager.cs

```diff
@@ -93,7 +91,10 @@ public void InitializeGameState()
         MAKE SURE TO USE THE INSTANCE PROPERTY AS THAT IS THE SINGLE INSTANCE
     */
     // Getters
-    public string GetNextLearningMinigameFilename() { return _nextLearningMinigameFilename; }
+    public string GetNextLearningMinigameFilename()
+    {
+        return _nextLearningMinigameSecurityConcept + GetAttackMinigamesAttemptsRequiredToUpgrade(_nextLearningMinigameSecurityConcept).ToString() + ".xml";
```

Comment:
Just trynna understand how `GetAttackMinigamesAttemptsRequiredToUpgrade()` works with this. I thought the amount of attempts before you can upgrade was going to be fixed at 2?

Reply:
Here in ```GameManager.GetNextLearningMinigameFilename()```, ```GetAttackMinigamesAttemptsRequiredToUpgrade()``` is used to get the upgrade level number that corresponds to the XML file. This will always match the attempts required to get an upgrade.

Let's say that player is currently at 0 upgrades and 123 attack minigame attempts. ```GetAttackMinigamesAttemptsRequiredToUpgrade()``` would return 1 as only 1 attack attempt is required to upgrade.

I don't think we ever discussed a hard cap on attack minigame attempts allowed per upgrade. I believe we said that we would discourage only doing one attack minigame with the heat mechanic and block repeated attack minigame attempst with the opp knowledge distribution